llvm/include/llvm/CodeGen/MachinePassManager.h

```diff
@@ -78,9 +78,9 @@ struct MachinePassModel
     auto PA = this->Pass.run(IR, AM);
 
     if constexpr (is_detected<has_get_set_properties_t, PassT>::value)
-      IR.getProperties().set(PassT::getSetProperties());
+      IR.getProperties().set(this->Pass.getSetProperties());
     if constexpr (is_detected<has_get_cleared_properties_t, PassT>::value)
-      IR.getProperties().reset(PassT::getClearedProperties());
+      IR.getProperties().reset(this->Pass.getClearedProperties());
```

Comment:
I should move them into a dedicated function, like `getMachineFunctionPassPreservedAnalyses`, so `PassT().run(MF, MFAM)` can work properly...